DIFM Lite: Adds button to skip design selection

aneeshd16 opened the pull request:
#### Changes proposed in this Pull Request

Context: pdh1Xd-gN-p2

* Adds a "Let us choose" button to allow users to skip the design picker screen.
* Adds `difm-lite-default` as a feature to the Russel theme. If the user clicks on the "Let Us Choose" button, the first theme with the `difm-lite-default` feature will be set on the site. (Currently, the list of themes is on the client-side, but it will be powered from the API in the future).
* Also adds a `difm_lite_let_us_choose_selected` option which will be displayed in the Blog RC. 

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->
* Optional - Apply D72586-code
* Start the flow at `/start/do-it-for-me`.
* Choose "New Site" on the first step.
* On the design picker step, confirm that you see the "Let us choose" button:
![image](https://user-images.githubusercontent.com/5436027/148342862-78c18321-7968-4cf5-b6ca-9e989012f01d.png)
* Click the "Let Us Choose" button.
* Go through the rest of the flow and complete the purchase.
* Confirm that the theme has been changed to Russel. You can do this by checking the Blog RC.

**Testing for regressions in the design picker**
* Navigate to `/start/setup-site/intent?siteSlug=<any site slug>`
* Click on "Start Building".
* On the design picker step, confirm that you are not able to see the "Let us choose" button.

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

aneeshd16 committed:
DIFM Lite: Adds button to skip design selection

aneeshd16 commented:
Pending: Fire event on "Let us choose" button click.

aneeshd16 committed:
Adds Tracks event for Let us choose button

dzver commented on client/signup/steps/design-picker/let-us-choose.tsx:
```diff
@@ -0,0 +1,67 @@
+		<LetUsChooseContainer>
+			<div>
+				{ translate(
+					"Can't decide? " + 'No problem, our experts can choose the perfect design for your site!'
```
Do we need to concatenate this string?

aneeshd16 replied:
The only reason is to make it easier for translators: https://github.com/Automattic/wp-calypso/blob/trunk/packages/i18n-calypso/README.md#L44

p-jackson replied:
I'm not sure how the `+` makes it easier for translators 🤔 The concat operator is allowed for code styling reasons, the long string still appears as a single value to translators. For example, [this long string that uses concatenation in the code](https://github.com/Automattic/wp-calypso/blob/294238af02e1f3e67205c27af040c3afcd94b2d7/client/blocks/edit-gravatar/index.jsx#L214) appears as [a single continuous string in glotpress](https://translate.wordpress.com/projects/wpcom/es/default/?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=231250&filters%5Btranslation_id%5D=9624258).

I tried removing the `+` and Prettier was happy with the resulting line length, so I think it's more idiomatic to not concat in this instance.

dzver reviewed (approved):
Makes sense and works for me. Left one minor question.

p-jackson commented on client/signup/steps/design-picker/index.jsx:
```diff
@@ -156,15 +165,24 @@ export default function DesignPickerStep( props ) {
 
+	function renderCategoriesFooter() {
+		return (
+			<div>
```
Nit pick: could we use a React fragment here instead? i.e. `<> ... </>`. That way we're not introducing an unnecessary layer of DOM nesting.

p-jackson commented on client/signup/steps/design-picker/let-us-choose.tsx:
```diff
@@ -0,0 +1,67 @@
+import styled from '@emotion/styled';
```
I think this is the first place in the signup framework that uses `emotion`. I was a bit worried it'd include a whole bunch of extra bytes to the payload, but it doesn't look like it has. `emotion` must already be subtly included somewhere else, so I guess it's ok to start using it for our step styling 🤷

p-jackson replied:
Turns out it's already used a lot in the `woocommerce-install` flow: https://github.com/Automattic/wp-calypso/blob/79e0795af6e0a63504a61a0191ac700814cb4197/client/signup/steps/woocommerce-install/components/support-card/index.tsx#L10

p-jackson reviewed (approved):
Have a couple of comments, but this change looks good to me 👍 

For reference, this is what the mobile layout looks like
<img width="498" alt="Screenshot 2022-01-13 at 1 49 16 PM" src="https://user-images.githubusercontent.com/1500769/149246154-488c2c1a-b205-4ddf-838f-4017f38f7e38.png">

aneeshd16 committed:
Review feedback: minor refactor

1. Changes `<div>` to React fragment to reduce DOM nesting.
2. Removes string concatenation since it was unnecessary.

aneeshd16 commented:
> Have a couple of comments, but this change looks good to me 👍

@p-jackson, thank you for your feedback. I appreciate it! I have added the suggested changes in 255358b. Good spot with the `<div> -> <>`!

> For reference, this is what the mobile layout looks like 

Thanks! @onuro, do you have any suggestions to fix the mobile layout here? I'll tackle that in a follow-up PR.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7082141

Thank you @aneeshd16 for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7082141).
